PyDynamic/model_estimation/fit_filter.py

```diff
@@ -87,92 +86,207 @@ def _fitIIR(
     return b, a
 
 
-def LSIIR(Hvals, Nb, Na, f, Fs, tau=0, justFit=False):
-    """Least-squares IIR filter fit to a given frequency response.
-    
-    This method uses Gauss-Newton non-linear optimization and pole
-    mapping for filter stabilization
-    
+def LSIIR(
+    Hvals: np.ndarray,
+    Nb: int,
+    Na: int,
+    f: np.ndarray,
+    Fs: float,
+    tau: Optional[int] = 0,
+    justFit: Optional[bool] = False,
+    verbose: Optional[bool] = True,
+    max_stab_iter: Optional[int] = 50,
+    inv: Optional[bool] = False,
+    UHvals: Optional[np.ndarray] = None,
+    mc_runs: Optional[int] = 1000,
+) -> Union[
+    Tuple[np.ndarray, np.ndarray, int], Tuple[np.ndarray, np.ndarray, int, np.ndarray]
+]:
+    """Least-squares (time-discrete) IIR filter fit to a given frequency response or
+    its reciprocal
+
+    For the forward problem this method uses Gauss-Newton non-linear optimization.
```

Comment:
@eichstaedtPTB Please correct the wording here to reflect an appropriate problem description.

Reply:
```suggestion
    For the forward problem this method calculates the IIR filter coefficients as a least-squares fit to the frequency response.
```

Reply:
@eichstaedtPTB Thanks for the suggestion so far. In addition the "forward/inverse problem"  formulation was in question. We agreed that this is not well formulated. Could you provide a suggestion for replacing the terms "forward problem" and "inverse problem"?

Reply:
```suggestion
    For the case of fitting an IIR filter model to a frequency response this method calculates the IIR filter coefficients as a least-squares fit.
```